processing/src/main/java/org/apache/druid/query/filter/Filter.java

```diff
@@ -29,12 +30,66 @@
 import org.apache.druid.segment.vector.VectorColumnSelectorFactory;
 
 import javax.annotation.Nullable;
+import java.util.Collections;
 import java.util.Map;
 import java.util.Set;
 
 @SubclassesMustOverrideEqualsAndHashCode
 public interface Filter
 {
+  /**
+   * Compute indexes and build a container {@link FilterBundle} to be used during
+   * {@link org.apache.druid.segment.Cursor} or {@link org.apache.druid.segment.vector.VectorCursor} creation, combining
+   * the computed outputs of {@link #getBitmapColumnIndex(ColumnIndexSelector)} as well as references to
+   * {@link #makeMatcher(ColumnSelectorFactory)} and {@link #makeVectorMatcher(VectorColumnSelectorFactory)}.
+   *
+   * Filters populating the {@link FilterBundle} container should only set the values which MUST be evaluated by the
+   * cursor. See {@link FilterBundle} for additional details.
+   */
+  default <T> FilterBundle makeFilterBundle(
+      ColumnIndexSelector columnIndexSelector,
+      BitmapResultFactory<T> bitmapResultFactory,
+      int selectionRowCount,
+      int totalRowCount,
+      boolean includeUnknown,
+      boolean allowPartialIndex
+  )
+  {
+    final BitmapColumnIndex columnIndex = getBitmapColumnIndex(columnIndexSelector);
+
+    final FilterBundle.IndexBundle indexBundle;
+    final boolean needMatcher;
+    if (columnIndex != null) {
+      final long bitmapConstructionStartNs = System.nanoTime();
+      T result = columnIndex.computeBitmapResult(bitmapResultFactory, selectionRowCount, totalRowCount, includeUnknown);
+      final long totalConstructionTimeNs = System.nanoTime() - bitmapConstructionStartNs;
+      if (result != null) {
+        ImmutableBitmap bitmap = bitmapResultFactory.toImmutableBitmap(result);
+        indexBundle = new FilterBundle.SimpleIndexBundle(
+            Collections.singletonList(new FilterBundle.IndexBundleInfo(toString(), bitmap.size(), totalConstructionTimeNs)),
```

Comment:
that's a good point about huge filter strings. It is super useful to have the breakdown of filter partitioning, but maybe I can think of a way to have safeguards against this case, either adding a new method to filters instead of directly relying on toString to has these safeguards, or make it lazy so that it only happens if debug logging is enabled, or something else.